plugins/osdn/osdn.go

```diff
@@ -207,6 +207,11 @@ func (oi *OsdnRegistryInterface) WriteNetworkConfig(network string, subnetLength
 	if err == nil {
 		if cn.Network == network && cn.HostSubnetLength == int(subnetLength) && cn.ServiceNetwork == serviceNetwork {
 			return nil
+		} else if cn.Network == network && cn.HostSubnetLength == int(subnetLength) && cn.ServiceNetwork == "" {
+			// Upgrade from 3.0.0
+			cn.ServiceNetwork = serviceNetwork
+			_, err = oi.oClient.ClusterNetwork().Update(cn)
```

Comment:
Adding upgrade pieces in the regular code path doesn't look clean to me.
Can't we do these upgrades as part of conversions (looks like there is no version change in this case) or as part of one time migration from 3.0 to 3.1 release?
@liggitt how are we handling upgrades without version change in other places?